ecl/hqlcpp/hqlckey.cpp

```diff
@@ -1240,10 +1240,18 @@ void KeyedJoinInfo::splitFilter(IHqlExpression * filter, SharedHqlExpr & keyTarg
 
 void HqlCppTranslator::buildKeyedJoinExtra(ActivityInstance & instance, IHqlExpression * expr, KeyedJoinInfo * info)
 {
-    //virtual IOutputMetaData * queryDiskRecordSize() = 0;  // Excluding fpos and sequence
+    //virtual IOutputMetaData * queryDiskRecordSize() = 0;
     if (info->isFullJoin())
-        buildMetaMember(instance.classctx, info->queryRawRhs(), false, "queryDiskRecordSize");
-
+    {
+        LinkedHqlExpr serializedRecord = info->queryOriginalKey()->queryRecord();
+        if (serializedRecord->hasAttribute(_payload_Atom))
+            serializedRecord.setown(notePayloadFields(serializedRecord->queryRecord(), numPayloadFields(serializedRecord)));
+        serializedRecord.setown(getSerializedForm(serializedRecord, diskAtom));
+
+        bool hasFilePosition = getBoolAttribute(serializedRecord, filepositionAtom, true);  // MORE - why true?
```

Comment:
true, because it defaults to having a fileposition field, unless they say fileposition(false)